File: Frontend/Certiva/src/app/Models/rol-dto.ts
export interface RolDTO {
  idRol: number;
  nombre: string;
  descripcion?: string;
  activo?: boolean;
  fechaCreacion?: string;
  fechaActualizacion?: string;
  /** Presente en {@code /api/roles/listar} y {@code /api/roles/registro} */
  codigo?: string;
}

File: Backend/api/src/main/java/com/certiva/api/DTO/Rol/RolDTO.java
package com.certiva.api.DTO.Rol;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RolDTO {

    private Long idRol;

    private String nombre;

    private String descripcion;

    private Boolean activo;

    private LocalDateTime fechaCreacion;

    private LocalDateTime fechaActualizacion;
}
